src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java

```diff
@@ -121,38 +125,93 @@ private void testNonExistentCourse() {
+    }
+
+    @Test
+    protected void testAccessControl_invalidParameterValues_shouldFail() {
         ______TS("non-existent course");
 
         InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
-        CourseAttributes courseAttributes = logic.getCourse(instructor1OfCourse1.getCourseId());
+        loginAsInstructor(instructor1OfCourse1.getGoogleId());
 
         String[] submissionParams = new String[] {
                 Const.ParamsNames.COURSE_ID, "not-exist",
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
         };
 
-        loginAsInstructor(instructor1OfCourse1.getGoogleId());
+        assertNull(logic.getCourse("not-exist"));
 
         verifyCannotAccess(submissionParams);
 
-        ______TS("Inaccessible without being an instructor or student");
+        ______TS("non-existent entitytype");
 
         submissionParams = new String[] {
-                Const.ParamsNames.COURSE_ID, courseAttributes.getId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
+                Const.ParamsNames.ENTITY_TYPE, "no-entity",
+        };
+
+        verifyCannotAccess(submissionParams);
+    }
+
+    @Test
+    protected void testAccessControl_testInstructorAccess_shouldPass() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.getCourseId(),
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR,
+        };
+
+        verifyOnlyInstructorsOfTheSameCourseCanAccess(submissionParams);
+    }
+
+    @Test
+    protected void testAccessControl_testStudentAccess_shouldPass() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
         };
 
+        verifyAccessibleForStudentsOfTheSameCourse(submissionParams);
         verifyInaccessibleWithoutLogin(submissionParams);
         verifyInaccessibleForUnregisteredUsers(submissionParams);
+        verifyInaccessibleForInstructors(submissionParams);
         verifyInaccessibleForAdmin(submissionParams);
+    }
+
+    @Test
+    protected void testAccessControl_onlyStudentOrInstructorAccessRequired_shouldPass()
+            throws InvalidParametersException, EnrollException, EntityDoesNotExistException,
+            EntityAlreadyExistsException {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        CourseAttributes typicalCourse2 = typicalBundle.courses.get("typicalCourse2");
+        logic.enrollStudents("Team|Section|Name|Email|Comment" + System.lineSeparator()
```

Comment:
I think you can just update a student's googleId to make it associated with the `instructor1OfCourse1.googleId`